### PR TITLE
Make bamtools_split output a dataset collection

### PR DESCRIPTION
**Pull request by pvanheus**
Replace the multiple output datasets of bamtools_split with a dataset collection output.

Also update the tests and the documentation to reflect this change.

### CODE REVIEW

**Comment by erasche** [comment]
Code all looks good, assuming tests pass.

File: tool_collections/bamtools/bamtools_split/bamtools-split.xml [hunk 1]
```diff
@@ -5,7 +5,6 @@
   </requirements>
   <command>
```
Comment (bgruening) [inline tool_collections/bamtools/bamtools_split/bamtools-split.xml]: You should increase the version number.

**Comment by pvanheus** [comment]
There was some discussion on the Hangout earlier about the different semantic modes of bamtools split. Turns out they are:

1) Split into mapped and unmapped reads
2) Split paired-end reads into two single end collections
3) Split by reference (split into one dataset for each reference contig)
4) Split by tag

Modes 1 and 2 split into exactly 2 datasets. Modes 3 and 4 split into an undefined number of dataset, with mode 3 potentially most useful for parallel processing. There does not (except for, arguably, mode 3) be a "split into chunks" options. The discussion revolved around the differences between "split into chunks" and other split modes and the possible need to create different tools to address the different semantics of these modes. I'd like to see some comments on this issue before a merge.

Secondly, this change introduces a undocumented feature (output to dataset collection) as default behaviour in a devteam tool. This might prove confusing to users: should this merge wait on documentation?

File: tool_collections/bamtools/bamtools_split/bamtools-split.xml [hunk 2]
```diff
@@ -54,20 +53,18 @@
...
             <param name="input_bam" ftype="bam" value="bamtools-input1.bam"/>
             <param name="analysis_type_selector" value="-mapped"/>
-            <output name="report">
-              <assert_contents>
-                <has_line line="BAM" />
-              </assert_contents>
-              <discovered_dataset designation="MAPPED" file="bamtools-split-test1.bam" ftype="bam"/>
-            </output>
+            <output_collection name="report">
+              <element name="MAPPED" file="bamtools-split-MAPPED1.bam" />
```
Comment (bgruening) [inline tool_collections/bamtools/bamtools_split/bamtools-split.xml]: Indentation ...

**Comment by bgruening** [comment]
@pvanheus can you add a few sentence to the documentation and a boolean "output to dataset collection" (default: no) for the next version. So we have a smooth transition. In the version after the next we will change the default to yes.

**Comment by bgruening** [comment]
ping @pvanheus

**Comment by bgruening** [comment]
ping @pvanheus

**Comment by bgruening** [comment]
ping @pvanheus

**Comment by davebx** [comment]
Implemented in and/or superseded by galaxyproject/tools-iuc#1359

**Pull request closed without merging**